### PR TITLE
start/stop pom commands 

### PR DESCRIPTION
**Pull request by jennschiffer**
as i look into handling conversation state, i'm putting in the conditional logic from the requirements/analysis part of the process so i can visualize it all better in the code 🎅

### CODE REVIEW

File: src/pombot/commands/start.js [hunk 1]
```diff
@@ -7,5 +7,22 @@ export default createCommand({
   name: 'start',
   description: 'Starts a pom timer.',
 }, () => {
-  return `> Theoretically I should be starting a pom right now.`;
+  // get pom status
+  const pomStatus = 'running';
+
+  switch (pomStatus) {
```
Comment (tkellen) [inline src/pombot/commands/start.js]: how about having a file called `states.js` like this:
```js
export default {
  RUNNING: 1,
  ON_BREAK: 2,
  NOT_RUNNING: 3,
};
```

...and sharing that?

Comment (jennschiffer) [inline src/pombot/commands/start.js]: i added this but i'm not sure why.

what i'm trying to figure out now is how to keep the timer to the current conversation. if i start a pom in a channel, and try to start one in dm, the bot says one is already going on

**Comment by cowboy** [comment]
It might be helpful to create a `Pom` class that you can create instances of, with all the behaviors you're implementing as methods on that class. That way, the bot can just be a UI layer on top of the class. Also, it would allow you to create multiple `Pom` instances, and also write bot-agnostic tests.

**Comment by jennschiffer** [comment]
@cowboy thanks, that's exactly what i'm reading up on now!

File: src/pombot/index.js [hunk 2]
```diff
@@ -23,21 +24,34 @@ const bot = createSlackBot({
...
-          stopCommand,
-          iwillCommand,
-          statusCommand,
-        ]),
+      // create the pom for this message
+      const pom = new Pom({
+        maxMinutes: 1,
+        warningMinutes: 0.5,
+        onWarningCallback: () => {
+          this.sendResponse({channel}, `warning: there are *${pom.timeLeft / 60000}* minutes left in this pom!`);
```
Comment (cowboy) [inline src/pombot/index.js]: `sendResponse` expects a first argument like `{channel: 'D0JJE9TBR'}` (the channel ID string).

Try `{channel: channel.id}` instead.

File: src/pombot/commands/start.js [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+    description: 'Starts a pom timer.',
+  }, message => {
+
+    switch (pom.state) {
+      case states.RUNNING:
+        // if pom is running, stay minutes left in current pom
+        return `there is already a pom running, there are *${pom.getMinutes(pom.timeLeft)} minutes* left.`;
+        break;
+      case states.ON_BREAK:
+        // if pom is on break, say minutes left in break
```
Comment (tkellen) [inline src/pombot/commands/start.js]: do we need this comment? i think you've composed this nicely enough that just reading the code itself says this comment

File: src/pombot/states.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*
+* all the states for a pom
+*/
+
+const states = {
+  RUNNING: 0,
```
Comment (cowboy) [inline src/pombot/states.js]: I like setting states to a string with their name like `RUNNING: 'RUNNING',` that way if you ever have to log the state, `RUNNING` is easier to comprehend than `0`.

File: src/pombot/pom.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  /* stops a pom and its timers */
+  stop() {
+    this.state = states.NOT_RUNNING;
+    this.startTime = null;
+    clearTimeout(this.onWarningId);
+    clearTimeout(this.onDoneId);
+  }
+
+  /* returns if the timer has started */
+  isStarted() {
```
Comment (cowboy) [inline src/pombot/pom.js]: Is this redundant if you have `states.RUNNING`? If not, should it use `states.RUNNING` internally?

Comment (tkellen) [inline src/pombot/pom.js]: :+1:

File: src/pombot/pom.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    return this.startTime !== null;
+  }
+
+  /* returns the time that has elapsed in milliseconds */
+  get timeElapsed() {
+    if (this.isStarted()) {
```
Comment (tkellen) [inline src/pombot/pom.js]: maybe overkill, but you could get some nice time display for free using moment

Comment (jennschiffer) [inline src/pombot/pom.js]: yeah i'd like to do that in a separate PR, cleaning messages up after all the functionality is set

Comment (tkellen) [inline src/pombot/pom.js]: :+1:
